ocs/cluster.py

```diff
@@ -0,0 +1,248 @@
+        self.mons = self._filter_pods(self.mon_selector, self.pods)
+        self.mdss = self._filter_pods(self.mds_selector, self.pods)
+        self.toolbox = self._filter_pods(self.tool_selector, self.pods)
+
+        # set port attrib on mon pods
+        self.mons = list(map(self.set_port, self.mons))
+
+    @staticmethod
+    def set_port(pod):
+        """
+        Set port attribute on pod
+        Args:
+            pod(Pod): Pod object
+        """
+        l1 = pod.pod_data.get('spec').get('containers')
+        l2 = l1[0]['ports'][0]['containerPort']
+        # Dynamically added attribute 'port'
+        pod.port = l2
+        logging.info(f"port={pod.port}")
+        return pod
+
+    def mon_change_count(self, new_count):
+        """
+        Change mon count in the cluster
+
+        Args:
+            new_count(int): Absolute number of mons required
+        """
+        self.cluster.data['spec']['mon']['count'] = new_count
+        self.cluster.apply(**self.cluster.data)
+        self.mon_health_check(new_count)
+        logger.info(f"Mon count changed to {new_count}")
+        self.mon_count = new_count
+        self.scan_cluster()
+        self.cluster.reload()
+
+    def mon_health_check(self, count):
+        """
+        TODO: Mon pod count alone can't tell monitor health
+            need other parameters as well
+        """
+        try:
+            assert POD.wait_for_resource(
+                condition='Running', selector=self.mon_selector,
+                resource_count=count,
+            )
+        except AssertionError as e:
+            logger.error(e)
+            raise MonCountException(
+                f"Failed to achieve desired Mon count"
+                f" {count}"
+            )
+
+    def mds_change_count(self, new_count):
+        """
+        Change mds count in the cluster
+
+        Args:
+            new_count(int): Absolute number of mds required
+        """
+        self.cephfs.data['spec']['metadataServer']['activeCount'] = new_count
+        logger.info(self.cephfs.data)
+        self.cephfs.apply(**self.cephfs.data)
+        self.mds_health_check(new_count)
+        logger.info(f"MDS count changed to {new_count}")
+        self.mds_count = new_count
+        self.scan_cluster()
+        self.cephfs.reload()
+
+    def mds_health_check(self, count):
+        """
+        TODO: mds count alone can't tell whether fs is healthy
+                need some more parameters as well
+        """
+        try:
+            assert POD.wait_for_resource(
+                condition='Running', selector=self.mds_selector,
+                resource_count=count
+        )
+        except AssertionError as e:
+            logger.error(e)
+            raise MDSCountException(
+                f"Failed to achieve desired MDS count"
+                f" {count}"
+            )
+
+    def get_admin_key(self):
+        """
+        Returns:
+            adminkey(str): base64 encoded key
```

Comment:
adminkey (str): base64 encoded key

Reply:
Sorry, I am not able to notice what is different here.

Reply:
Done, I figured it as space missing.